docs/VnVPlan/VnVPlan.tex

```diff
@@ -209,55 +209,488 @@ \section{System Test Description}
+          
+Output: Database contains correctly organized items.
+
+Test Case Derivation: System back-end must accurately map and sort receipt entries provided, keeping a record of each item all users purchase.
+          
+How test will be performed: Begin with a populated testing database. Test cases comprised of items and prices are sent through an API call to the system back-end by testing client. Then an additional API call will be called from the testing client against the system back-end to validate that the items and prices are correctly stored in the database.
+
+\item{\textbf{FRT-FR7-3}}
+
+Control: Automatic, Dynamic
+          
+Initial State: Test database is populated.
+
+Input: API calls from client (front-end) to back-end containing receipt data. Data containing items that are the same as existing items in the database but with slightly different names.
+          
+Output: Database contains correctly organized items and understanding the relationship between like items.
+
+Test Case Derivation: System back-end must accurately map and sort receipt entries provided, keeping a record of each item all users purchase.
+          
+How test will be performed: Begin with a populated testing database. Test cases comprised of items and prices are sent through an API call to the system back-end by testing client. Then an additional API call will be called from the testing client against the system back-end to validate that the items and prices are correctly stored in the database.
+
+\end{enumerate}
+
+\paragraph{Grocery Recommendations}
+
+These tests pertain to all the functional requirements related to the systems grocery recommendations based on grocery data received from user and their preferences.
+
+\begin{enumerate}
+
+\item{\textbf{FRT-FR8-1}}
+
+Control: Manual, Dynamic
+          
+Initial State: User has set budget information.
+
+Input: User submits receipt containing luxury items that are out of budget.
+          
+Output: System provides recommendation to stop purchasing some items that are too expensive and (if available) recommend alternatives that are more in-line with the users budget.
+
+Test Case Derivation: System must be aware of the users purchasing activity and recommend the user with cheaper items that are similar.
+          
+How test will be performed: Tester will submit luxury items that exceed the users set budget. Tester will validate that the system recommends the user with cheaper alternatives to the luxury items.
+
+\item{\textbf{FRT-FR8-2}}
+
+Control: Manual, Dynamic
+
+How test will be performed: Survey is presented to the user for them to respond how useful the system recommendations are. A success is determined by at least a 70\% positive response. The survey question is detailed in \textit{section 6.2.2}.
+
+\item{\textbf{FRT-FR9-1}}
+
+Control: Manual, Dynamic
+          
+Initial State: User finds submits that same item for cheap price from location X.
```

Comment:
The database has an entry for a grocery item that is purchased from a location X.